src/neuroconv/tools/roiextractors/roiextractors.py

```diff
@@ -270,45 +243,60 @@ def add_imaging_plane_to_nwbfile(
     ----------
     nwbfile : NWBFile
         An previously defined -in memory- NWBFile.
-    metadata : dict
-        The metadata in the nwb conversion tools format.
-    imaging_plane_name: str
-        The name of the imaging plane to be added.
+    metadata : dict, optional
+        The metadata in the nwb conversion tools format. If None, uses defaults from _get_default_ophys_metadata().
+    imaging_plane_name: str, optional
+        The name of the imaging plane to be added. If None, uses the default imaging plane name.
 
     Returns
     -------
     NWBFile
         The nwbfile passed as an input with the imaging plane added.
     """
-
-    # Set the defaults and required infrastructure
-    metadata_copy = deepcopy(metadata)
     default_metadata = _get_default_ophys_metadata()
-    metadata_copy = dict_deep_update(default_metadata, metadata_copy, append_list=False)
-    add_devices_to_nwbfile(nwbfile=nwbfile, metadata=metadata_copy)
-
-    default_imaging_plane_name = default_metadata["Ophys"]["ImagingPlane"][0]["name"]
-    imaging_plane_name = imaging_plane_name or default_imaging_plane_name
-    existing_imaging_planes = nwbfile.imaging_planes
-
-    if imaging_plane_name not in existing_imaging_planes:
-        imaging_plane_metadata = next(
-            (
-                imaging_plane_metadata
-                for imaging_plane_metadata in metadata_copy["Ophys"]["ImagingPlane"]
-                if imaging_plane_metadata["name"] == imaging_plane_name
-            ),
-            None,
-        )
-        if imaging_plane_metadata is None:
-            raise ValueError(
-                f"Metadata for Imaging Plane '{imaging_plane_name}' not found in metadata['Ophys']['ImagingPlane']."
-            )
+    default_imaging_plane = default_metadata["Ophys"]["ImagingPlane"][0]
+
+    imaging_plane_name = imaging_plane_name or default_imaging_plane["name"]
+
+    if imaging_plane_name in nwbfile.imaging_planes:
+        return nwbfile
 
-        imaging_plane = _create_imaging_plane_from_metadata(
-            nwbfile=nwbfile, imaging_plane_metadata=imaging_plane_metadata
+    add_devices_to_nwbfile(nwbfile=nwbfile, metadata=metadata)
+
+    metadata = metadata or {}
+    imaging_planes_list = metadata.get("Ophys", {}).get("ImagingPlane", [])
+
+    # Search for imaging plane by name in user metadata
+    user_plane_metadata = next(
+        (plane for plane in imaging_planes_list if plane["name"] == imaging_plane_name),
+        None,
+    )
+
+    # If user provided imaging planes but the requested name was not found, raise an error
+    has_user_requested_plane = user_plane_metadata is not None
+    has_imaging_planes = bool(imaging_planes_list)
+    if not has_user_requested_plane and has_imaging_planes:
+        raise ValueError(
+            f"Metadata for Imaging Plane '{imaging_plane_name}' not found in metadata['Ophys']['ImagingPlane']."
```

Comment:
I'm finding this block of code a little bit hard to read, maybe just because I'm tired, but I think the variable names might be a bit clearer.

```python
if imaging_plane_name is not None and imaging_plane_name_is_not_in_metadata:
...
```
I think, if I'm reading correctly, this is what the if statement is trying to accomplish here?

Reply:
Thanks for the feedback, the process is the following:
1) the user provides an imaging_plane_name
2) We search the metadata to see if there is an entry whose name is `imaging_plane_name`.
3) if this is not find there, raise an error.

A more verbose version of the code is:

```python
    metadata_has_user_requested_plane = user_plane_metadata is not None
    metadata_has_imaging_planes_metadata = bool(imaging_planes_list)
    if not metadata_has_user_requested_plane and metadata_has_imaging_planes_metadata:
       raise error telling the user that they give us metadata but give us a name that is not there.
``` 

I had more verbose names like those at the beginning but it is my October wish to be less verbose x D

If you have name suggestions let me know.

Reply:
Ah!

I think you are right.

We should throw an error when the user requests a name for which no metadata entry exists, right?

This is easier to read and does not have that but that you mention.

Am I getting you right?

Reply:
I made some changes on that direction, check them out. I am probably tired too.

Reply:
Wait, actually, I don't think that metadata should ever be None at this level. It is a function to build and add the imaging_plane from the metadata so metadata should always be passed.

It might happen that the metadata entry that the user request does not exist in the passed metadata but we already have a test for that:

https://github.com/catalystneuro/neuroconv/blob/5504e0a34829b131d9c1541231657bc6e8e3469d/tests/test_modalities/test_ophys/test_tools_roiextractors.py#L237-L246